src/compat/libc/string/memccpy.c

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file
+ * @brief Implementation of #memccpy() function.
+ * 
+ * @date 18/3/20
+ * @author Giannis Zapantis (github name: tech-gian)
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+void* memccpy(void* dest, const void* src, int c, size_t n) {
+
+/* If overlap happens the results
+ * will be undefined.
+ */
+
+
+/* if there is already data after c
+ * then that data stays the same.
+ * if there isn't we don't change
+ * anything.
+ */
+
+/* copying dest to temp_dest
+ * and src to temp_dest.
+ * So we can work for each
+ * byte separately.
+*/
+
+    char* temp_dest = (char*) dest;
+    char* temp_src = (char*) src;
+    size_t pos = n;
+
+    char* ar1[n];
+
+/* getting each byte for c. */
+    char temp_c[4];
+    temp_c[0] = (c >> 24) & 0xFF;
+    temp_c[1] = (c >> 16) & 0xFF;
+    temp_c[2] = (c >> 8) & 0xFF;
+    temp_c[3] = c & 0xFF;
+
+
+/* if c!=0 then we check the last byte.
+ * if there is in src.
+ */
+if (c!=0) {

```

Comment:
What is the weird formatting
Use tabs at the start of the lines please

Reply:
https://github.com/embox/embox/pull/1894/commits/1fb0ae2368e6c6fa61a094bdeec09d23563959a4

This is what you mean with tabs at the start of the lines?

Reply:
No
I mean [this part of Embox code-style](https://github.com/embox/embox/wiki/Code-Style#tabs-and-brackets)

Reply:
Oh, I'm really sorry. I didn't even noticed that. Thank you :) I corrected it. Is there anything else, that need to be changed? I want to do a very good work, even for this "small" function.